### PR TITLE
feat: stricter rules when generating user ids

### PR DESCRIPTION
**Pull request by danielgent**
~Wrangler's `nanoid` is breaking everything. Doesn't actually have the functions needed to configure it. TBD~
I can't test this new function as everything explodes there (maybe some module issue)

*BUT* this actually works on Cloudflare :smile: 

See a new record I just made. Id - `85b1ada7f1b932c51aaf2b05`

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,6 +57,7 @@
     "kysely-planetscale": "^1.4.0",
     "liquidjs": "^10.9.2",
     "mime": "^3.0.0",
+    "nanoid": "^3.3.3",
```
Comment (danielgent) [inline package.json:60]: we are referencing this so it _should_ be in our package.json! :smile: 

This version matches the wrangler version. It works on cloudflare anyway :muscle:

File: src/authentication-flows/passwordless.ts [hunk 2]
```diff
@@ -24,8 +23,7 @@ export async function validateCode(env: Env, params: LoginParams) {
   let user = await env.data.users.getByEmail(client.tenant_id, params.email);
   if (!user) {
     user = await env.data.users.create(client.tenant_id, {
-      // TODO: replace nanoid with some hash of email
-      id: `email|${nanoid()}`,
```
Comment (danielgent) [inline src/authentication-flows/passwordless.ts:28]: I'm also removing the `provider` concatenation here

On the next task I'll do I'll put this back when getting the user object - https://app.shortcut.com/sesamy/story/4903/top-lever-user-id-is-concatenation-of-provider-user-id

File: test/authentication-flows/ticket.spec.ts [hunk 3]
```diff
@@ -8,6 +8,14 @@ import {
 } from "../../src/types";
 import { parseJwt } from "../../src/utils/parse-jwt";
 
+// nanoid blows up in Jest when try to import customAlphabet...
+jest.mock("nanoid", () => {
```
Comment (markusahlstrand) [inline test/authentication-flows/ticket.spec.ts:12]: I had this before.. If you check in the __mocks__ folder we're doing this:
```
// Nanois is a mess with esm an modules..
module.exports = {
  nanoid: () => "testid",
};
```

**Review by markusahlstrand — APPROVED** [review]
Looks good!

File: test/authentication-flows/ticket.spec.ts (hunk 3 above)
Comment (danielgent) [reply test/authentication-flows/ticket.spec.ts:12]: Ahhhhhh.  Damn ok you'd already solved this. I should just have added this alphabet export here

**Pull request merged**